MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -4595,6 +4595,20 @@ public void cancelMothballOrActivation() {
         return crew;
     }
 
+    /**
+     * Prototype TSM makes a unit harder to repair and maintain.
+     *
+     * @return Whether the unit has prototype TSM
+     */
+    public boolean hasPrototypeTSM() {
+        for (Mounted m : getEntity().getMisc()) {
+            if (m.getType().hasFlag(MiscType.F_TSM) && m.getType().hasFlag(MiscType.F_PROTOTYPE)) {
```

Comment:
One of these days, we'll need to come up with a mechanism for checking if a unit has some specific piece of equipment other than looping through all the damn equipment and looking for flags.

Reply:
For the record, I don't like the implementation here of increased repair time either, but I don't know a better way.